source/matplot/core/axes_type.cpp

```diff
@@ -426,7 +426,7 @@ namespace matplot {
             run_command("set polar");
         }
         auto set_or_unset_axis = [this](class axis_type &ax,
-                                        std::string axis_name,
+                                        const std::string &axis_name,
```

Comment:
If that was your concern about breaking the API, both `std::string axis_name` and `std::string const& axis_name` are OK. In the implementation the `std::string axis_name` can be moved and the `std::string const& axis_name` can be copied. As far as the user is concerned, that doesn't functionally break the API. Only `std::string &axis_name` would be a problem.

Reply:
No concerns here, that just a lambda expression - internal stuff.
And yes, it'll bind to the same calls as previously, we just need to make sure that the lambda expression itself does not modify the parameters, but that is checked by the compiler.